tanner/sessions/session_analyzer.py

```diff
@@ -18,18 +18,20 @@ def __init__(self, loop=None):
         self.logger = logging.getLogger('tanner.session_analyzer.SessionAnalyzer')
         self.attacks = ['sqli', 'rfi', 'lfi', 'xss', 'php_code_injection', 'cmd_exec', 'crlf']
 
-    async def analyze(self, session_key, redis_client, pg_client):
-        session = None
+    async def analyze(self, redis_client, pg_client):
+        sessions = None
         await asyncio.sleep(1, loop=self._loop)
         try:
-            session = await redis_client.get(session_key, encoding='utf-8')
-            session = json.loads(session)
+            keys = await redis_client.keys("[0-9a-f]*")
         except (aioredis.ProtocolError, TypeError, ValueError) as error:
             self.logger.exception('Can\'t get session for analyze: %s', error)
         else:
-            result = await self.create_stats(session, redis_client)
-            await self.queue.put(result)
-            await self.save_session(redis_client, pg_client)
+            for key in keys:
+                session = await redis_client.get(key, encoding='utf-8')
+                session = json.loads(session)
+                result = await self.create_stats(session, redis_client)
+                await self.queue.put(result)
+                await self.save_session(redis_client, pg_client)
```

Comment:
do we really need a separate save function or we can do everything in current function?

anyway, save should not be called after you put each session into the queue, it should be called once